pandas/core/indexes/datetimelike.py

```diff
@@ -342,7 +342,7 @@ def _format_with_header(self, header, **kwargs):
     def __contains__(self, key):
         try:
             res = self.get_loc(key)
-            return is_scalar(res) or type(res) == slice or np.any(res)
+            return is_scalar(res) or type(res) == slice or np.any(res + 1)
```

Comment:
can you change ``type(res) == slice`` to ``isinstance(res, slice)``.

I think condition could be (for the last clause)
``is_list_like(res) and len(res)``. just need to account for an indexer, not the character of it